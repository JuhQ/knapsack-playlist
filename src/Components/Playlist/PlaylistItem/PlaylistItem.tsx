import React from "react";
import { Item } from "semantic-ui-react";

import { YoutubeItem } from "../../../types";
import Duration from "../../Duration/Duration";

interface Props {
  item: YoutubeItem;
  onClick: (item: YoutubeItem) => void;
}

const PlaylistItem: React.FC<Props> = ({ item, onClick }: Props) => (
  <Item as="a" onClick={() => onClick(item)}>
    <Item.Image src={`https://i.ytimg.com/vi/${item.id}/hqdefault.jpg`} />
    <Item.Content>
      <Item.Header>{item.title}</Item.Header>
      <Item.Meta>
        <Duration seconds={item.seconds} />
      </Item.Meta>
    </Item.Content>
  </Item>
);

export default PlaylistItem;
