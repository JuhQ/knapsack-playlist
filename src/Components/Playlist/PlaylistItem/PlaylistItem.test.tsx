import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import PlaylistItem from "./PlaylistItem";

describe("PlaylistItem", () => {
  it("should render", () => {
    const component = renderer.create(
      <PlaylistItem
        item={{ id: "1", title: "test", seconds: 60 }}
        onClick={() => null}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be able to call onClick function", () => {
    const handleOnClick = jest.fn();
    const wrapper = shallow(
      <PlaylistItem
        item={{ id: "1", title: "test", seconds: 60 }}
        onClick={handleOnClick}
      />
    );
    expect(handleOnClick).not.toHaveBeenCalled();
    wrapper.simulate("click");
    expect(handleOnClick).toHaveBeenCalled();
  });
});
