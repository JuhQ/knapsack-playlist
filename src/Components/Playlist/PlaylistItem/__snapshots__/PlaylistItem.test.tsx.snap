// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistItem should render 1`] = `
<a
  className="item"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      src="https://i.ytimg.com/vi/1/hqdefault.jpg"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      test
    </div>
    <div
      className="meta"
    >
      1 minute
       
      
    </div>
  </div>
</a>
`;
